### ISSUE TITLE
Default::duration missing version number in std documentation

### ISSUE THREAD

**Issue by jethrogb** [rust-lang/rust#40480]
`impl Default for Duration` was [added in 1.15](https://github.com/rust-lang/rust/pull/37699) but the [current documentation](https://doc.rust-lang.org/std/time/struct.Duration.html) doesn't indicate it wasn't available before that.

**Comment by steveklabnik** [comment]
We never talk about old versions for documentation; if you want to see that version's docs, check out the docs for that version.

Thanks, but this is something we just don't do.

**Comment by jethrogb** [comment]
I'm not sure what you mean, there are version numbers all over the page I linked.

**Comment by steveklabnik** [comment]
I don't see "version" anywhere there, could you send me a screenshot or something? It's possible I'm just misunderstanding you.

**Comment by steveklabnik** [comment]
Ohhhh I see now, you're not talking about in text, you're talking about the stability flags.

I believe that implementations are auto-stable, and therefore, don't have the flag, and therefore, don't have this. So this is more of a feature request.

**Comment by jethrogb** [comment]
![screen shot 2017-03-13 at 11 40 14-fullpage](https://cloud.githubusercontent.com/assets/1132307/23870019/59a67ab2-07e2-11e7-80f7-8ef27e4ad748.png)

**Comment by jethrogb** [comment]
The `XyzAssign` implementations have a version number though.

**Comment by jmesmon** [comment]
Likely because the trait itself was stabilized on that version number (`XyzAssign` were added after 1.0.0).

Agreed that having a marker on impls would be useful.

**Comment by jethrogb** [comment]
Ok, looking at the source, it seems the cause is that you can't specify a stability attribute for a  `#[derive]`. Related #29152

**Comment by steveklabnik** [comment]
Triage: no changes I'm aware of.

**Comment by jyn514** [comment]
I don't think this is really a rustdoc bug - the information just isn't available in the source code. I'm not sure who's in charge of `#[stable]` and `#[unstable]`, maybe T-compiler?

That said, I think something like `#[derive(Clone(stable = "1.15.0"))]` might work.

**Comment by petrochenkov** [comment]
>the information just isn't available in the source code

It should be available, starting from https://github.com/rust-lang/rust/pull/29152 stability attributes are copied from the item to derived impls.
If they are not displayed in the doc, then something goes wrong along the way.

**Comment by petrochenkov** [comment]
https://play.rust-lang.org/?version=nightly&mode=debug&edition=2018&gist=9ef33271d10b380c800731eb31835323

```rust
#[stable(feature = "name", since = "version")]
#[derive(Clone)]
struct S;
```
```
#[stable(feature = "name", since = "version")]
struct S;
#[automatically_derived]
#[allow(unused_qualifications)]
#[stable(feature = "name", since = "version")]
impl ::core::clone::Clone for S {
    #[inline]
    fn clone(&self) -> S { match *self { S => S, } }
}
```

**Comment by jyn514** [comment]
Hmm, that seems like the wrong behavior though? `Duration` was stabilized in 1.3.0 but `impl Default for Duration` is only present since 1.15.0. I would rather not show the stability than show the wrong stability.

**Comment by petrochenkov** [comment]
Ah, I see what is the issue now, the reported stabilization version for the impl is incorrect exactly because it's inherited from the item.

I don't think it's something important enough to extend derive syntax for (or do anything else really).
